### PR TITLE
[AIRFLOW-XXX] Add How-To-Guide to GCP PubSub

### PR DESCRIPTION
**Pull request by TobKed**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: airflow/gcp/operators/pubsub.py [hunk 1]
```diff
@@ -572,6 +588,10 @@ def execute(self, context):
 class PubSubPublishOperator(BaseOperator):
     """Publish messages to a PubSub topic.
 
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
```
Comment (nuclearpinguin) [inline airflow/gcp/operators/pubsub.py]: I am not sure about the comma, @mschickensoup WDYT?

File: docs/howto/operator/gcp/pubsub.rst [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    :end-before: [END howto_operator_gcp_pubsub_args_common]
+
+
+.. _howto/operator:PubSubTopicCreateOperator:
+
+Create a PubSub topic
+^^^^^^^^^^^^^^^^^^^^^
+
+The PubSub topic is a named resource to which messages are sent by publishers.
+The :class:`~airflow.gcp.operators.pubsub.PubSubTopicCreateOperator` operator would create a topic.
```
Comment (nuclearpinguin) [inline docs/howto/operator/gcp/pubsub.rst]:
```suggestion
The :class:`~airflow.gcp.operators.pubsub.PubSubTopicCreateOperator` operator creates a topic.
```

File: airflow/gcp/operators/pubsub.py (hunk 1 above)
Comment (TobKed) [reply airflow/gcp/operators/pubsub.py]: This sentence is used widely with comma 🤔 . Is it proper sentence @mschickensoup ?

Comment (mschickensoup) [reply airflow/gcp/operators/pubsub.py]: It's correct, feel free to leave it as it is :-) -> https://developers.google.com/style/cross-references?hl=vi

**Comment by TobKed** [comment]
PR to apache created: https://github.com/apache/airflow/pull/6497

File: docs/howto/operator/gcp/pubsub.rst [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+Google Cloud PubSub Operators
+=============================
+
+`Google Cloud PubSub <https://cloud.google.com/pubsub/>`__ is a fully-managed real-time
+messaging service that allows you to send and receive messages between independent applications.
+You can leverage Cloud Pub/Sub’s flexibility to decouple systems and components hosted
+on Google Cloud Platform or elsewhere on the Internet.
+
+Publisher applications can send messages to a topic and other applications can subscribe to that topic to receive the messages.
+By decoupling senders and receivers, Google Cloud PubSub allows developers to communicate between independently written applications.
```
Comment (mschickensoup) [inline docs/howto/operator/gcp/pubsub.rst]:
```suggestion
By decoupling senders and receivers, Google Cloud PubSub allows developers to communicate between independently written applications.
```
By decoupling senders and receivers Google Cloud PubSub allows developers to communicate between independently written applications.

File: docs/howto/operator/gcp/pubsub.rst [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+.. include:: _partials/prerequisite_tasks.rst
+
+.. _howto/operator:PubSubTopicCreateOperator:
+
+Creating a PubSub topic
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The PubSub topic is a named resource to which messages are sent by publishers.
+The :class:`~airflow.providers.google.cloud.operators.pubsub.PubSubTopicCreateOperator` operator would create a topic.
```
Comment (mschickensoup) [inline docs/howto/operator/gcp/pubsub.rst]:
```suggestion
The :class:`~airflow.providers.google.cloud.operators.pubsub.PubSubTopicCreateOperator` operator would create a topic.
```
The :class:`~airflow.providers.google.cloud.operators.pubsub.PubSubTopicCreateOperator` operator creates a topic.

Comment (mschickensoup) [reply docs/howto/operator/gcp/pubsub.rst]: https://developers.google.com/style/tense?hl=vi

File: docs/howto/operator/gcp/pubsub.rst [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    :start-after: [START howto_operator_gcp_pubsub_create_topic]
+    :end-before: [END howto_operator_gcp_pubsub_create_topic]
+
+
+.. _howto/operator:PubSubSubscriptionCreateOperator:
+
+Creating a PubSub subscription
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The subscription is named resource representing the stream of messages from a single, specific topic,
```
Comment (mschickensoup) [inline docs/howto/operator/gcp/pubsub.rst]: Is this subscription any kind of subscription or you're referring to the specific one?

File: docs/howto/operator/gcp/pubsub.rst [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+to be delivered to the subscribing application.
+The :class:`~airflow.providers.google.cloud.operators.pubsub.PubSubSubscriptionCreateOperator` operator would create a subscription.
```
Comment (mschickensoup) [inline docs/howto/operator/gcp/pubsub.rst]:
```suggestion
The :class:`~airflow.providers.google.cloud.operators.pubsub.PubSubSubscriptionCreateOperator` operator  creates the subscription.
```

File: docs/howto/operator/gcp/pubsub.rst [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    :start-after: [START howto_operator_gcp_pubsub_create_subscription]
+    :end-before: [END howto_operator_gcp_pubsub_create_subscription]
+
+
+.. _howto/operator:PubSubPublishOperator:
+
+Publishing PubSub messages
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A ``Message`` is the combination of data and (optional) attributes that a publisher sends to a topic and is eventually delivered to subscribers.
```
Comment (mschickensoup) [inline docs/howto/operator/gcp/pubsub.rst]:
```suggestion
A ``Message`` is a combination of data and (optional) attributes that a publisher sends to a topic and is eventually delivered to subscribers.
```

File: docs/howto/operator/gcp/pubsub.rst [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+    :language: python
+    :start-after: [START howto_operator_gcp_pubsub_publish]
+    :end-before: [END howto_operator_gcp_pubsub_publish]
+
+
+.. _howto/operator:PubSubPullSensor:
+
+Pulling messages from a PubSub subscription
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The :class:`~airflow.providers.google.cloud.sensors.pubsub.PubSubPullSensor` sensor would pull messages from a PubSub subscription
```
Comment (mschickensoup) [inline docs/howto/operator/gcp/pubsub.rst]:
```suggestion
The :class:`~airflow.providers.google.cloud.sensors.pubsub.PubSubPullSensor` sensor pulls messages from a PubSub subscription
```

Comment (mschickensoup) [reply docs/howto/operator/gcp/pubsub.rst]: Any PubSub subscription?

File: docs/howto/operator/gcp/pubsub.rst [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+and pass them through XCom.
+
+.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_pubsub.py
+    :language: python
+    :start-after: [START howto_operator_gcp_pubsub_pull_message]
+    :end-before: [END howto_operator_gcp_pubsub_pull_message]
+
+To pull messages from XCom :class:`~airflow.operators.bash_operator.BashOperator` operator would do this.
```
Comment (mschickensoup) [inline docs/howto/operator/gcp/pubsub.rst]: I'd change the whole sentence. Something like this ```To pull messages from XCom use the following pattern / example / code etc.```

File: docs/howto/operator/gcp/pubsub.rst [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+    :start-after: [START howto_operator_gcp_pubsub_pull_messages_result]
+    :end-before: [END howto_operator_gcp_pubsub_pull_messages_result]
+
+
+.. _howto/operator:PubSubSubscriptionDeleteOperator:
+
+Deleting a PubSub subscription
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The :class:`~airflow.providers.google.cloud.operators.pubsub.PubSubSubscriptionDeleteOperator` operator would delete a subscription.
```
Comment (mschickensoup) [inline docs/howto/operator/gcp/pubsub.rst]:
```suggestion
The :class:`~airflow.providers.google.cloud.operators.pubsub.PubSubSubscriptionDeleteOperator` operator  deletes the subscription.
```
Please, don't use `would`:3 BTW, are you again referring to the specific subscription in lights of the class you mentioned?

File: docs/howto/operator/gcp/pubsub.rst [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+    :start-after: [START howto_operator_gcp_pubsub_unsubscribe]
+    :end-before: [END howto_operator_gcp_pubsub_unsubscribe]
+
+
+.. _howto/operator:PubSubTopicDeleteOperator:
+
+Deleting a PubSub topic
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The :class:`~airflow.providers.google.cloud.operators.pubsub.PubSubTopicDeleteOperator` operator would delete a topic.
```
Comment (mschickensoup) [inline docs/howto/operator/gcp/pubsub.rst]:
```suggestion
The :class:`~airflow.providers.google.cloud.operators.pubsub.PubSubTopicDeleteOperator` operator deletes a topic.
```
No `would` pls.

**Pull request closed without merging**